extensions/wikia/WikiaHubsServices/models/WikiaHubsModel.class.php

```diff
@@ -24,29 +24,17 @@ public function getVertical() {
 	}
 
 	public function getVerticalName($verticalId) {
-		/** @var WikiFactoryHub $wikiFactoryHub */
-		$wikiFactoryHub = WikiFactoryHub::getInstance();
-		$wikiaHub = $wikiFactoryHub->getCategory($verticalId);
-		return wfMessage('hub-' . $wikiaHub['name'])->inContentLanguage()->text();
+		return WikiFactoryHub::getInstance()->getVerticalNameMessage( $verticalId )->inContentLanguage()->text();
 	}
 
 	public function getCanonicalVerticalName($verticalId) {
```

Comment:
@kvas-damian Would it make any sense to make a common method getVerticalName( $verticalId, $isCanonical=false) to remove a bit of redundancy or it doesn't make any sense in this context?